tests/torch/fx/test_models.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+import torch
+import torch.fx
+import torch.nn.parallel
+import torch.optim
+import torch.utils.data
+import torch.utils.data.distributed
+import torchvision.models as models
+from torch._export import capture_pre_autograd_graph
+from ultralytics.models.yolo import YOLO
+
+from nncf.common.graph.graph import NNCFNodeName
+from nncf.common.graph.operator_metatypes import OperatorMetatype
+from nncf.common.utils.os import safe_open
+from nncf.experimental.torch.fx.nncf_graph_builder import GraphConverter
+from nncf.torch.dynamic_graph.patch_pytorch import disable_patching
+from nncf.torch.graph.graph import PTNNCFGraph
+from tests.shared.paths import TEST_ROOT
+from tests.torch.test_compressed_graph import check_graph
+
+
+@pytest.fixture(name="fx_dir")
+def fx_dir_fixture(request):
+    fx_dir_name = "fx"
+    return fx_dir_name
+
+
+@dataclass
+class ModelCase:
+    model_id: str
+    input_shape: Tuple[int]
+
+
+MODELS = (
+    ModelCase("resnet18", (1, 3, 224, 224)),
+    ModelCase("mobilenet_v3_small", (1, 3, 224, 224)),
+    ModelCase("vit_b_16", (1, 3, 224, 224)),
+    ModelCase("swin_v2_s", (1, 3, 224, 224)),
+    ModelCase("yolov8", (1, 3, 224, 224)),
+)
+
+
+def get_model(model_id: str, device: torch.device) -> torch.nn.Module:
+    if model_id == "yolov8":
+        model = YOLO("yolov8n.yaml")
+        model = model.model
+    else:
+        model = getattr(models, model_id)(weights=None)
+    model.to(device)
+
+    return model
+
+
+def get_dot_filename(model_name):
+    return model_name + ".dot"
+
+
+def get_json_filename(model_name):
+    return model_name + ".json"
+
+
+def get_full_path_to_json(model_json_name: str) -> str:
+    path_to_dir = TEST_ROOT / "torch" / "data" / "reference_graphs" / "fx" / "reference_metatypes"
+    path_to_json = path_to_dir / model_json_name
+    return path_to_json
+
+
+def check_to_create_ref(metatype_dict: Dict[NNCFNodeName, Type[OperatorMetatype]], json_dir: str):
+    json_parent_dir = Path(json_dir).parent
+
+    if os.getenv("NNCF_TEST_REGEN_JSON") is not None:
+        if not os.path.exists(json_parent_dir):
+            os.makedirs(json_parent_dir)
+        write_ref_metatype(metatype_dict, Path(json_dir))
+
+
+def write_ref_metatype(metatype_dict: Dict[NNCFNodeName, Type[OperatorMetatype]], path: Path) -> None:
+    with safe_open(path, "w") as file:
+        json.dump(metatype_dict, file)
+
+
+def read_json_file(path: Path):
+    with safe_open(path, "r") as file:
+        model_ref_metatypes = json.load(file)
+    return model_ref_metatypes
+
+
+def get_ref_metatypes_from_json(
+    model_name: str, model_metatypes: Dict[NNCFNodeName, Type[OperatorMetatype]]
+) -> Dict[NNCFNodeName, Type[OperatorMetatype]]:
+    model_json_name = get_json_filename(model_name)
+    complete_path = get_full_path_to_json(model_json_name)
+    check_to_create_ref(model_metatypes, complete_path)
+    model_ref_metatypes = read_json_file(complete_path)
+    return model_ref_metatypes
```

Comment:
Please decrease amount of functions, like this:
```suggestion


def get_ref_metatypes_from_json(
    model_name: str, model_metatypes: Dict[NNCFNodeName, Type[OperatorMetatype]]
) -> Dict[NNCFNodeName, Type[OperatorMetatype]]:
    model_json_name = get_json_filename(model_name)
    complete_path = get_full_path_to_json(model_json_name)
    
    json_parent_dir = Path(json_dir).parent

    if os.getenv("NNCF_TEST_REGEN_JSON") is not None:
        if not os.path.exists(json_parent_dir):
            os.makedirs(json_parent_dir)
        with safe_open(path, "w") as file:
        json.dump(metatype_dict, file)
    
     with safe_open(path, "r") as file:
        return json.load(file)
```